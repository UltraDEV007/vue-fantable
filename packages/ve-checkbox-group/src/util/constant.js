// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
