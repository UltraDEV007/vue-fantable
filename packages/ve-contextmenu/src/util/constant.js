// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
