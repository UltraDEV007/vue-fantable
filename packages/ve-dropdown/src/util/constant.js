// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
