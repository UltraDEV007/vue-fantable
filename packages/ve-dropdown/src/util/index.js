/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
