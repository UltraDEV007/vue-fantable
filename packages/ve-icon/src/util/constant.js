// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
