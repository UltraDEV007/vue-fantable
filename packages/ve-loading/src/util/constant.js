// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
