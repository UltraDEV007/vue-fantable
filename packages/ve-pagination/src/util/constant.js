// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
