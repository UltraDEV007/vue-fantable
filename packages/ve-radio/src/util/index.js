import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
