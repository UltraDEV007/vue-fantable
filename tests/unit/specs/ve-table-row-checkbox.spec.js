import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable row checkbox", () => {
    const TABLE_DATA = [
        {
            rowKey: 1001,
            name: "John",
            date: "1900-05-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            rowKey: 1002,
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            rowKey: 1003,
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            rowKey: 1004,
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            rowKey: 1005,
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    const COLUMNS = [
        {
            field: "",
            key: "a",
            // type=checkbox
            type: "checkbox",
            title: "",
            width: 50,
            align: "center",
        },
        {
            field: "name",
            key: "b",
            title: "Name",
            width: 200,
            align: "left",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            width: 300,
            align: "left",
        },
        {
            field: "address",
            key: "d",
            title: "Address",
            width: "",
            align: "left",
        },
    ];

    it("render", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {},
                rowKeyFieldName: "rowKey",
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("has checkbox", () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {},
                rowKeyFieldName: "rowKey",
            },
        });

        expect(wrapper.find(".ve-table-header-tr .ve-checkbox").exists()).toBe(
            true,
        );
        expect(wrapper.find(".ve-table-body-tr .ve-checkbox").exists()).toBe(
            true,
        );
    });

    it("check default selected key", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    defaultSelectedRowKeys: [1001, 1003, 1004],
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        expect(
            wrapper
                .find(
                    ".ve-table-header-tr .ve-checkbox .ve-checkbox-indeterminate",
                )
                .exists(),
        ).toBe(true);

        // checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(3);
    });

    it("check disable selected keys with defaultSelectedRowKeys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    // 禁用的选择（禁止勾选或者禁止取消勾选）
                    disableSelectedRowKeys: [1002, 1005],
                    // 默认选择
                    defaultSelectedRowKeys: [1001, 1003, 1004, 1005],
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        expect(
            wrapper
                .find(
                    ".ve-table-header-tr .ve-checkbox .ve-checkbox-indeterminate",
                )
                .exists(),
        ).toBe(true);

        // checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(4);

        // disable checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked.ve-checkbox-disabled",
            ).length,
        ).toBe(1);

        // disable unchecked count
        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(1)
                .find(".ve-checkbox-content")
                .classes()
                .includes("ve-checkbox-checked"),
        ).toBe(false);
    });

    it("check disable selected keys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    // 禁用的选择（禁止勾选或者禁止取消勾选）
                    disableSelectedRowKeys: [1002, 1005],
                    // 默认选择
                    //defaultSelectedRowKeys: [1001, 1003, 1004, 1005],
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        expect(
            wrapper
                .find(
                    ".ve-table-header-tr .ve-checkbox .ve-checkbox-indeterminate",
                )
                .exists(),
        ).toBe(false);

        // checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(0);

        // disable checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked.ve-checkbox-disabled",
            ).length,
        ).toBe(0);

        // disable unchecked count
        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(1)
                .find(".ve-checkbox-content")
                .classes()
                .includes("ve-checkbox-checked"),
        ).toBe(false);
    });

    it("select all event with check disable selected keys", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    // 禁用的选择（禁止勾选或者禁止取消勾选）
                    disableSelectedRowKeys: [1002, 1005],
                    // 默认选择
                    //defaultSelectedRowKeys: [1001, 1003, 1004, 1005],
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        // select all
        wrapper
            .findAll(".ve-table-header-tr")
            .at(0)
            .findAll(".ve-table-header-th")
            .at(0)
            .find(".ve-checkbox")
            .trigger("click");

        await later();

        expect(
            wrapper
                .find(
                    ".ve-table-header-tr .ve-checkbox .ve-checkbox-indeterminate",
                )
                .exists(),
        ).toBe(true);

        // checked count
        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(3);

        // disable unchecked count
        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(1)
                .find(".ve-checkbox-content")
                .classes()
                .includes("ve-checkbox-checked"),
        ).toBe(false);

        expect(
            wrapper
                .findAll(".ve-table-body-tr")
                .at(4)
                .find(".ve-checkbox-content")
                .classes()
                .includes("ve-checkbox-checked"),
        ).toBe(false);
    });

    it("controllable attr selectedRowKey", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    // 可控属性
                    selectedRowKeys: [1003],
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        expect(
            wrapper
                .find(
                    ".ve-table-header-tr .ve-checkbox .ve-checkbox-indeterminate",
                )
                .exists(),
        ).toBe(true);

        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(1);

        wrapper.setProps({
            checkboxOption: {
                selectedRowKeys: [1003, 1004],
            },
        });

        await later();

        expect(
            wrapper.findAll(
                ".ve-table-body-tr .ve-checkbox .ve-checkbox-checked",
            ).length,
        ).toBe(2);
    });

    it("header hideSelectAll", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    // 可控属性
                    hideSelectAll: true,
                },
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        expect(wrapper.find(".ve-table-header-tr .ve-checkbox").exists()).toBe(
            false,
        );
    });

    it("checkboxOption selectedRowChange event", async () => {
        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    selectedRowChange: ({
                        row,
                        isSelected,
                        selectedRowKeys,
                    }) => {
                        mockFn(row, isSelected, selectedRowKeys);
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .findAll(".ve-table-body-td")
            .at(0)
            .find(".ve-checkbox")
            .trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledWith(TABLE_DATA[0], true, [1001]);
    });

    it("checkboxOption selectedAllChange event", async () => {
        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                checkboxOption: {
                    selectedAllChange: ({ isSelected, selectedRowKeys }) => {
                        mockFn(isSelected, selectedRowKeys);
                    },
                },
                rowKeyFieldName: "rowKey",
            },
        });

        wrapper
            .find(".ve-table-header-tr")
            .findAll(".ve-table-header-th")
            .at(0)
            .find(".ve-checkbox")
            .trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledWith(
            true,
            [1001, 1002, 1003, 1004, 1005],
        );
    });
});
