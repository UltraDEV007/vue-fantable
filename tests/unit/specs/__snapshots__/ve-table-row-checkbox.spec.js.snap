// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veTable row checkbox > render 1`] = `
"<div class="vue-table-root">
  <div style="width: 100%;" class="fan-table fan-table-border-around">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <!---->
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 200px;">
            <col style="width: 300px;">
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header" tableviewportwidth="0">
            <tr class="fan-table-header-tr">
              <th style="text-align: center; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Name
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Hobby
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th fan-table-last-column" rowspan="1" colspan="1">
                <!---->Address
                <!---->
                <!---->
                <!---->
              </th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-stripe fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr" row-key="1001">
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="a">
                <!----><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->John
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Shanghai
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="1002">
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="a">
                <!----><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->Dickerson
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Beijing
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="1003">
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="a">
                <!----><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->Larsen
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Chongqing
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="1004">
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="a">
                <!----><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->Geneva
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Xiamen
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="1005">
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="a">
                <!----><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;" column="[object Object]"><span class="ve-checkbox-content"><input class="ve-checkbox-input" type="checkbox"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"><!----></span></label>
                <!---->
                <!---->
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->Jami
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Shenzhen
              </td>
            </tr>
            <!---->
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-middle">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-fixed-right">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;" class="fan-table-edit-input-container" allrowkeys="1001,1002,1003,1004,1005"><textarea class="fan-table-edit-input" tabindex="-1"></textarea></div>
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`veTable row checkbox render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 200px;">
            <col style="width: 300px;">
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="1001" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr row-key="1002" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr row-key="1003" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr row-key="1004" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr row-key="1005" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: center;"><label class="ve-checkbox fan-table-checkbox-wrapper" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value=""><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label"></span></label></td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: left;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
